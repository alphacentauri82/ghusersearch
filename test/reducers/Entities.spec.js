import entitiesReducer from 'reducers/Entities';

describe('Reducer: entities', () => {

  describe('when passed no state', () => {
    it('should return the initial state', () => {
      expect(
        entitiesReducer(undefined, {})
      ).toMatchSnapshot();
    });
  });

  describe('when action has an entities property', () => {
    it('should merge the object into the state', () => {
      const action = {
        payload: {
          entities: {
            users: {
              123: {name: 'bar'},
              456: {name: 'foo'},
            },
          },
        },
      };
      const beforeState = {
        users: {
          789: {name: 'baz'},
        },
        other: {foo: 'bar'},
      };
      const afterState = entitiesReducer(beforeState, action);

      expect(
        afterState
      ).toMatchSnapshot();
      expect(afterState).not.toBe(beforeState);
    });
  });

});
