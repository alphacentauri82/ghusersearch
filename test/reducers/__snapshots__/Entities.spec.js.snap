// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reducer: entities when action has an entities property should merge the object into the state 1`] = `
Object {
  "other": Object {
    "foo": "bar",
  },
  "users": Object {
    "123": Object {
      "name": "bar",
    },
    "456": Object {
      "name": "foo",
    },
    "789": Object {
      "name": "baz",
    },
  },
}
`;

exports[`Reducer: entities when passed no state should return the initial state 1`] = `
Object {
  "repos": Object {},
  "users": Object {},
}
`;
